File: src/main/java/gargoyle/ct/log/LoggerFactoryConfig.java
package gargoyle.ct.log;

import gargoyle.ct.log.jul.JULLoggerFactory;

import java.io.FileNotFoundException;
import java.io.IOException;
import java.io.InputStream;

public final class LoggerFactoryConfig {

    public static final String MSG_NOT_FOUND = "configuration {0} not found";
    static final String LOGGING_PROPERTIES = "/config/logging.properties";

    private LoggerFactoryConfig() {
    }

    public static Class<? extends ILoggerFactory> getLoggerFactoryClass() {
        return JULLoggerFactory.class;
    }

    public static InputStream getConfiguration() throws IOException {
        InputStream stream = LoggerFactory.class.getResourceAsStream(LOGGING_PROPERTIES);
        if (stream == null)
            throw new FileNotFoundException(LOGGING_PROPERTIES);
        return stream;
    }
}

File: src/main/kotlin/gargoyle/ct/log/LoggerFactoryConfig.kt
package gargoyle.ct.log

import gargoyle.ct.log.jul.JULLoggerFactory
import java.io.FileNotFoundException
import java.io.IOException
import java.io.InputStream
import kotlin.reflect.KClass

object LoggerFactoryConfig {
    const val MSG_NOT_FOUND = "configuration {0} not found"
    const val LOGGING_PROPERTIES = "/config/logging.properties"
    val loggerFactoryClass: KClass<out ILoggerFactory>
        get() = JULLoggerFactory::class

    @get:Throws(IOException::class)
    val configuration: InputStream
        get() = LoggerFactory::class.java.getResourceAsStream(LOGGING_PROPERTIES)
            ?: throw FileNotFoundException(LOGGING_PROPERTIES)
}
